cmd/bots/bots.go

```diff
@@ -33,45 +32,49 @@ func run() error {
 		webhookLocalAddress string // only for webhook method
 		publicURL           string // only for webhook method
 		botToken            string
+		storagePath         string
 	)
 	flag.StringVar(&nanomsgURL, "nano-msg-url", "ipc:///tmp/nano-msg-nodemon-pubsub.ipc", "Nanomsg IPC URL. Default is tcp://:8000")
 	flag.StringVar(&behavior, "behavior", "webhook", "Behavior is either webhook or polling")
 	flag.StringVar(&webhookLocalAddress, "webhook-local-address", ":8081", "The application's webhook address is :8081 by default")
 	flag.StringVar(&botToken, "bot-token", "", "Temporarily: the default token is the current token")
 	flag.StringVar(&publicURL, "public-url", "", "Default is https://mainnet-go-htz-fsn1-1.wavesnodes.com/bot")
+	flag.StringVar(&storagePath, "storage", "", "Path to storage")
 	flag.Parse()
 
-	if botToken == "" || publicURL == "" {
-		log.Println("Invalid bot token or public URL")
+	if botToken == "" {
+		log.Println("Invalid bot token")
+		return errors.New("invalid parameters")
+	}
+	if behavior == config.WebhookMethod && publicURL == "" {
+		log.Println("invalid public url")
+		return errors.New("invalid parameters")
+	}
+	if storagePath == "" {
+		log.Println("invalid storage path")
 		return errors.New("invalid parameters")
 	}
 
 	ctx, done := signal.NotifyContext(context.Background(), os.Interrupt)
 	defer done()
 
-	botSettings, err := config.NewBotSettings(behavior, webhookLocalAddress, publicURL, botToken)
+	tgBotEnv, err := tgBot.InitTgBot(behavior, webhookLocalAddress, publicURL, botToken, storagePath)
 	if err != nil {
-		return errors.Wrap(err, "failed to set up bot configuration")
+		log.Println("failed to initialize telegram bot")
+		return errors.Wrap(err, "failed to init tg bot")
 	}
-	bot, err := tele.NewBot(*botSettings)
-	if err != nil {
-		return errors.Wrap(err, "failed to start bot")
-	}
-
-	messagingEnv := &messaging.MessageEnvironment{ReceivedChat: false}
-	handlers.InitHandlers(bot, messagingEnv)
-
+	bots := messaging.NewBots(tgBotEnv)
 	go func() {
-		err := messaging.StartMessagingClient(ctx, nanomsgURL, bot, messagingEnv)
+		err := messaging.StartMessagingClient(ctx, nanomsgURL, bots)
 		if err != nil {
 			log.Printf("failed to start messaging service: %v", err)
 			return
 		}
 	}()
 
-	log.Println("Telegram bot started")
-	bot.Start()
+	for _, bot := range bots {
+		bot.Start()
+	}
```

Comment:
I think one app for all bot isn't good idea. It would be better if you create new app for each bot.